### PR TITLE
adding convolved response function

### PR DESCRIPTION
**Pull request by Jay4869**
making a python file for producing convolving response

Test: just run `python convolving_response.py` will produce the graph about experiment and neuronal signal and `conv.txt`

Note: I have already set parameter `n_volx = 173`, but we change it if necessary

### CODE REVIEW

**Comment by amandazhang** [comment]
Oops, we repeat each other's work again. In the constructing_convo function, you write:

neural_prediction = events2neural(fname + '_cond.txt', 2.5, n_volx), 

however the real data files we have are in the form "condxxx.txt" which is different from what was given as practice by instructor.

Thus you should modify that part.

**Comment by Jay4869** [comment]
So, you mean `cond_ds114_sub009_t2r1.txt` looks like this pattern?

**Comment by Jay4869** [comment]
Okay! I found those condition files
```
./model/model001/onsets/task002_run001/cond001.txt
./model/model001/onsets/task002_run001/cond005.txt
./model/model001/onsets/task002_run001/cond007.txt
./model/model001/onsets/task002_run001/cond004.txt
./model/model001/onsets/task002_run001/cond003.txt
./model/model001/onsets/task002_run001/cond002.txt
./model/model001/onsets/task002_run001/cond006.txt
./model/model001/onsets/task003_run001/cond001.txt
./model/model001/onsets/task003_run001/cond005.txt
./model/model001/onsets/task003_run001/cond007.txt
./model/model001/onsets/task003_run001/cond004.txt
./model/model001/onsets/task003_run001/cond003.txt
./model/model001/onsets/task003_run001/cond002.txt
./model/model001/onsets/task003_run001/cond006.txt
./model/model001/onsets/task001_run001/cond001.txt
./model/model001/onsets/task001_run001/cond005.txt
./model/model001/onsets/task001_run001/cond007.txt
./model/model001/onsets/task001_run001/cond004.txt
./model/model001/onsets/task001_run001/cond003.txt
./model/model001/onsets/task001_run001/cond002.txt
./model/model001/onsets/task001_run001/cond006.txt
```

**Comment by amandazhang** [comment]
You should have a look on the pull request I made early this afternoon, there's a path for you to get 'ds114_blah_cond.txt'like file we practiced during class from data file we downloaded.

**Comment by amandazhang** [comment]
Yes, you made it~

**Comment by amandazhang** [comment]
Also, you write:
plt.plot(all_tr_times, neural_prediction)
plt.plot(all_tr_times, convolved)
plt.show()

in the second function,  but you didn't output the plot image in the function. So what's the meaning of adding these three lines inside of the function?

**Comment by lizeyuyuz** [comment]
Looking at the condition txt files, good thing that the duration for each onset is the same (meaning, the intervals are evenly spaced), but we need to figure out which of the conditions are useful.

**Comment by amandazhang** [comment]
Good point Zeyu, and Jay just asked about what are those cond.txt file in each task file. Do you have any idea on that?

**Comment by Jay4869** [comment]
Let me change our code can handle on the general cases.

**Comment by lizeyuyuz** [comment]
@amandazhang  I think he already found the files, in case you want to see them, they are in the model file (mode-> model001-> onsets-> task001_run001)

**Comment by amandazhang** [comment]
@lizeyuyuz Oh that's not what I mean. I'm making a further question on issue you raised, "we need to figure out which of the conditions are useful". What do you mean by that

**Comment by Jay4869** [comment]
Do you have any preference name of convolving response file? Just like `conv001.txt`?

**Comment by lizeyuyuz** [comment]
@Jay4869 conv001.txt is the only response file for subject001? Maybe specify which task it is for.

**Comment by amandazhang** [comment]
@Jay4869 I think we can use pattern as in lectures, at least we should put 'taskXXX' to distinguish btw different tasks.

**Comment by lizeyuyuz** [comment]
@Jay4869 @amandazhang 
I looked at the 7 cond00#.txt files. I think cond002 and cond003 make the most sense. cond002 and cond003 only differ by the 3rd column, which is the amplitude(which we will not use for convolved.txt). So I say we just use cond002.

File: convoluving_response.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    undershoot_values = gamma.pdf(times, 12)
+    # Combine them
+    values = peak_values - 0.35 * undershoot_values
+    # Scale max to 0.6
+    return values / np.max(values) * 0.6
+
+def constructing_convo(fname, n_volx):
+	tr_times = np.arange(0, 30, 2.5)
+	hrf_at_trs = hrf(tr_times)
+	neural_prediction = events2neural(fname + '_cond.txt', 2.5, n_volx)
```
Comment (lizeyuyuz) [inline convoluving_response.py]: I just realized we can comment code! That's pretty cool. Anyways, you need to change the input file names. The cond.txt files for our data under the model/model001/onsets are named like cond001.txt

Comment (Jay4869) [inline convoluving_response.py]: check out the new vision which already fix it!

**Comment by amandazhang** [comment]
@lizeyuyuz Just like you said, the 2nd column is the duration. Thus I think it doesn't matter whether it's integer or not, nor does it matter the duration time. But I'm still confused on which of the 7 cond file to use and why not use all of them?

**Comment by amandazhang** [comment]
We should be careful on using function hrf, since those non-parametric numbers like 6,12,0.35,0.6 shouldn't always be those values for all cases right?

**Comment by Jay4869** [comment]
New updating convoluving.py
Add get_n_volx function to get the number of n_volx because our n_volx does not have to be 173, should be followed shape of `bold.nii.gz`

Please everyone check it, if there is any concern, please comment at below

**Pull request closed without merging**